editor/src/components/context-menu-items.ts

```diff
@@ -341,32 +343,29 @@ export const insert: ContextMenuItem<CanvasData> = {
   name: 'Add Element…',
   shortcut: 'A',
   enabled: true,
-  action: (data, dispatch) => {
-    requireDispatch(dispatch)([
-      setFocus('canvas'),
-      EditorActions.openFloatingInsertMenu({
-        insertMenuMode: 'insert',
-        parentPath: null,
-        indexPosition: null,
-      }),
-    ])
+  action: (data, _dispatch, _coord, event) => {
+    // FIXME Render prop support
```

Comment:
is this fixme for a future PR?

Reply:
Crud, this was a reminder for me to fix this as part of this PR. Though technically render prop support isn't something that would make sense for "Add Element", it certainly is required for "Swap Element"

Reply:
Ok change of plan, I'm going to merge this PR as is and create follow up issues so that the remaining work here can be parallelised